extra_data/cli/make_virtual_cxi.py

```diff
@@ -107,9 +99,26 @@ def main(argv=None):
 
     log.info("Reading run directory %s", run_dir)
     run = RunDirectory(run_dir, inc_suspect_trains=(not args.exc_suspect_trains))
-    det = _get_detector(run, args.min_modules)
-    if det is None:
-        sys.exit(f"No {_detectors()} sources found in {run_dir}")
+
+    _, det_class = identify_multimod_detectors(run, single=True)
+    det_name = det_class.__name__
+
+    min_modules = args.min_modules
+    if min_modules is None:
+        if det_name in ['AGIPD1M', 'DSSC1M', 'LPD1M']:
+            min_modules = 9
+        elif det_name in ['AGIPD500K']:
+            min_modules = 5
+        else:
+            min_modules = 1
```

Comment:
Rather than hardcoding class names here, maybe we can use the number of modules from the class if available - something like:

```python
n_modules = getattr(det_class, 'n_modules', 0)
min_modules = (n_modules // 2) + 1
```

Reply:
Thank you, implemented in https://github.com/European-XFEL/EXtra-data/commit/cc73dadb190a2c3e831a1fb08259f2752871ee5d.